ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs

```diff
@@ -0,0 +1,141 @@
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()).Result;
+        }
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsByAuthor()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    Author = "Norman Paulk"
+                }).Result;
+        }
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsByCommitter()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    Committer = "Fabrikamfiber16@hotmail.com"
+                }).Result;
+        }
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsOnABranch()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    ItemVersion = new GitVersionDescriptor()
+                    {
+                        VersionType = GitVersionType.Branch,
+                        VersionOptions = GitVersionOptions.None,
+                        Version = "master"
+                    }
+                }).Result;
+        }
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsOnABranchAndInAPath()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    ItemVersion = new GitVersionDescriptor()
+                    {
+                        VersionType = GitVersionType.Branch,
+                        VersionOptions = GitVersionOptions.None,
+                        Version = "master"
+                    },
+                    ItemPath = "/debug.log"
+                }).Result;
+        }
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsInDateRange()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    FromDate = "6/14/2018 6:19",
```

Comment:
It's a little odd that `GitQueryCommitsCriteria` uses `String` for FromDate and ToDat. I would consider having the sample  show how to construct a `DateTime` from a string (assuming this is how you'd normally create a DateTime from some string date value) and then convert this into a standard ISO 8601 formatted date/time value, which presumably the server accepts/understands. Feels more deterministic and gives the client full control.